cpu/stm32f1/include/cpu_conf.h

```diff
@@ -36,12 +37,122 @@ extern "C" {
 #endif
 
 /**
+ * @brief   Flash page configuration
+ * @{
+ */
+#define FLASHPAGE_SIZE      (2048U)
+
+#if defined(CPU_MODEL_STM32F103CB) || defined(CPU_MODEL_STM32F103RB)
+#define FLASHPAGE_NUMOF     (64U)
+#elif defined(CPU_MODEL_STM32F103RE)
+#define FLASHPAGE_NUMOF     (256U)
+#endif
+/** @} */
+
+/**
+ * @brief   Offset to reset handler on VTOR
+ * @{
+ */
+#define VTOR_RESET_HANDLER     0x4 /** One pointer after the beginning */
+/** @} */
+
+#if defined(CPU_MODEL_STM32F103RE)
+/*
+ * @brief   Flash partitioning for FW slots
+ * @{
+ */
+
+#ifndef FW_METADATA_SPACE
+#define FW_METADATA_SPACE          (0x100)
+#endif
+
+#define MAX_FW_SLOTS               (2)
+#define FW_SLOT_PAGES              (120)
+#define BOOTLOADER_SPACE           (0x4000)
+#define FW_SLOT_SIZE               FLASHPAGE_SIZE * FW_SLOT_PAGES
+#define FW_SLOT_1                  FLASH_BASE + BOOTLOADER_SPACE
+#define FW_SLOT_1_END              FW_SLOT_1 + FW_SLOT_SIZE
+#define FW_SLOT_1_PAGE             (8)
+#define FW_SLOT_2                  FW_SLOT_1_END
+#define FW_SLOT_2_END              FW_SLOT_2 + FW_SLOT_SIZE
+#define FW_SLOT_2_PAGE             (128)
```

Comment:
Same as for the Makefile, I suggest to use FW_SLOT(x) to allow flexibility. Maybe you find a `FW_SLOT(x)` which can be used to compute all of this.